src/custom/CustomButton.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+const CustomButton = ({
+  children,
+  dataCy = 'button',
+  clickHandler,
+  link = false,
+  to,
+}) => {
+  
+  const button = (
+    <button
+      className='custom-button'
+      data-cy={dataCy}
+      onClick={() => clickHandler}>
+      {children}
+    </button>
+  )
+
+  return <>{link ? <Link to={to}>{button}</Link> : button}</>
+}
+
+export default CustomButton
```

Comment:
damn! I don't understand the code. What is children for?

Reply:
Everything inside the CustomButton is sent down as props within the key `children` :D
```
<CustomButton>
  // children goes here
</CustomButton>
```